### PR TITLE
minor code cleaning

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

Fixes #\<issue_number>

## Before submitting

- [ ] Was this **discussed/approved** via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/metrics/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure to **update the docs**?
- [x] Did you write any new **necessary tests**?

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Review by krshrimali — APPROVED** [review]
Neat! :)

**Review by stancld — APPROVED** [review]

File: .github/assistant.py [hunk 1]
```diff
@@ -59,7 +59,7 @@ def prune_packages(req_file: str, *pkgs: str) -> None:
             lines = [ln for ln in lines if not ln.startswith(pkg)]
         logging.info(lines)
 
-        with open(req_file, "w") as fp:
+        with open(req_file, "w", encoding="utf_8") as fp:
```
Comment (justusschock) [inline .github/assistant.py:62]:
```suggestion
        with open(req_file, "w", encoding="utf-8") as fp:
```

as done by the docs here: https://docs.python.org/3/howto/unicode.html

File: .github/assistant.py [hunk 2]
```diff
@@ -71,17 +71,17 @@ def set_min_torch_by_python(fpath: str = "requirements.txt") -> None:
         with open(fpath) as fp:
             req = fp.read()
         req = re.sub(r"torch>=[\d\.]+", f"torch>={LUT_PYTHON_TORCH[py_ver]}", req)
-        with open(fpath, "w") as fp:
+        with open(fpath, "w", encoding="utf_8") as fp:
```
Comment (justusschock) [inline .github/assistant.py:74]:
```suggestion
        with open(fpath, "w", encoding="utf-8") as fp:
```

File: .github/assistant.py [hunk 3]
```diff
@@ -71,17 +71,17 @@ def set_min_torch_by_python(fpath: str = "requirements.txt") -> None:
...
             fp.write(req)
 
     @staticmethod
     def replace_min_requirements(fpath: str) -> None:
         """Replace all `>=` by `==` in given file."""
         logging.info(f"processing: {fpath}")
-        with open(fpath) as fp:
+        with open(fpath, encoding="utf_8") as fp:
```
Comment (justusschock) [inline .github/assistant.py:81]:
```suggestion
        with open(fpath, encoding="utf-8") as fp:
```

File: .github/assistant.py [hunk 4]
```diff
@@ -71,17 +71,17 @@ def set_min_torch_by_python(fpath: str = "requirements.txt") -> None:
...
             req = fp.read()
         req = req.replace(">=", "==")
-        with open(fpath, "w") as fp:
+        with open(fpath, "w", encoding="utf_8") as fp:
```
Comment (justusschock) [inline .github/assistant.py:84]:
```suggestion
        with open(fpath, "w", encoding="utf-8") as fp:
```

File: docs/source/conf.py [hunk 5]
```diff
@@ -65,7 +65,7 @@ def _transform_changelog(path_in: str, path_out: str) -> None:
         elif ln.startswith("### "):
             ln = ln.replace("###", f"### {chlog_ver} -")
             chlog_lines[i] = ln
-    with open(path_out, "w") as fp:
+    with open(path_out, "w", encoding="utf_8") as fp:
```
Comment (justusschock) [inline docs/source/conf.py:68]:
```suggestion
    with open(path_out, "w", encoding="utf-8") as fp:
```

File: torchmetrics/functional/text/squad.py [hunk 6]
```diff
@@ -133,7 +133,7 @@ def _squad_input_check(
     _fn_answer = lambda tgt: dict(
         answers=[dict(text=txt) for txt in tgt["answers"]["text"]], id=tgt["id"]  # type: ignore
     )
-    targets_dict = [dict(paragraphs=[dict(qas=[_fn_answer(target) for target in targets])])]
+    targets_dict = [{"paragraphs": [dict(qas=[_fn_answer(target) for target in targets])]}]
```
Comment (justusschock) [inline torchmetrics/functional/text/squad.py:136]:
```suggestion
    targets_dict = [{"paragraphs": [{"qas":[_fn_answer(target) for target in targets]}]}]
```

**Review by justusschock — APPROVED** [review]

**Review by SkafteNicki — APPROVED** [review]

**Pull request merged**